### PR TITLE
[GPU] Grouped decompression scale/zp support

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Added support for grouped decompression scale/zero point 
 - Added bfyx->fbyx conversion for decompression parameters to improve memory access pattern

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl [hunk 1]
```diff
@@ -193,7 +193,28 @@ KERNEL(fc)(
                 ACCUMULATOR_TYPE* w = (ACCUMULATOR_TYPE*)(&wei);
                 unroll_for(uint kii = 0; kii < TILE_K; ++kii) {
                     unroll_for(uint fi = 0; fi < TILE_OFM; ++fi) {
-                        w[kii * TILE_OFM + fi] = (w[kii * TILE_OFM + fi] - dzp[fi]) * ds[fi];
+                        const uint w_idx = kii * TILE_OFM + fi;
+                        uint offset_ofm = out_f + fi*SIMD + get_sub_group_local_id();
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl:197]:
```suggestion
                        const uint offset_ofm = out_f + fi*SIMD + sglid ;
```
And maybe it makes sense to move `w_idx` and `offset_ofm` calculations outside TILE_OFM loop?

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bfyx_ref.cl [hunk 2]
```diff
@@ -67,19 +72,24 @@ KERNEL(fc)(
            for (uint x = 0; x < INPUT0_SIZE_X; ++x)
             {
                 const uint input0_idx = INPUT0_GET_INDEX(b, ifm, y, x);
-                const uint filter_idx = GET_FILTER_INDEX(FILTER, 0, ofm, ifm, y, x);
                 #if COMPRESSED_WEIGHTS
-                    FILTER_TYPE filter_compressed = weights[filter_idx];
                     #if DECOMPRESSION_ZP_TERM
-                        ACCUMULATOR_TYPE zp = decompression_zp[DECOMPRESSION_ZP_GET_INDEX_SAFE(0, ofm, 0, 0)];
+                        ACCUMULATOR_TYPE zp = TO_ACCUMULATOR_TYPE(decompression_zp[DECOMPRESSION_ZP_GET_INDEX_SAFE(ofm, ifm / DECOMPRESSION_SCALE_GROUP_SIZE, 0, 0)]);
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bfyx_ref.cl:77]: Shouldn't we use `DECOMPRESSION_ZP_GROUP_SIZE` here and in other zero points-related calculations?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bfyx_ref.cl]: Fixed in https://github.com/openvinotoolkit/openvino/pull/20491/commits/88932f73cd69f052f0e8937d8e898c253a9e2bac

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl (hunk 1 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl]: Fixed in https://github.com/openvinotoolkit/openvino/pull/20491/commits/88932f73cd69f052f0e8937d8e898c253a9e2bac

> And maybe it makes sense to move w_idx and offset_ofm calculations outside TILE_OFM loop?

Those indices depend on `fi`. If you mean that we can move part of the offset compute outside of OFM loop, then looks like compiler does that.

**Pull request merged**